csp_billing_adapter/archive.py

```diff
@@ -45,3 +55,48 @@ def append_metering_records(
         return archive[1:]
     else:
         return archive
+
+
+def archive_record(
+    hook,
+    config: Config,
+    billing_record: dict
+) -> None:
+    """
+    :param hook:
+        The Pluggy plugin manager hook that will be
+        used to call the meter_billing operation.
+    :param config:
+        The configuration specifying the metrics that
+        need to be processed in the usage records list.
+    :param billing_record:
+        The dictionary containing the most recent
+        metering and usage records to be archived.
+    """
+    archive = retry_on_exception(
+        functools.partial(
+            hook.get_metering_archive,
+            config=config,
+        ),
+        logger=log,
+        func_name="hook.get_metering_archive"
+    )
+
+    if archive is None:
+        archive = []
```

Comment:
Note that this line isn't exercised by the unit tests right now, and the use of memory_archive.copy() in memory_archive.py's get_metering_archive() implementation means you can't just initialise it to None in cba_pm() in tests/unit/conftest.py.

Not a major issue, but a little tweaking to them memory_archive.py implementation could allow 100% code coverage ;-)

Reply:
Expectation is that plugins return property empty value from `get_metering_archive`, i.e. `[]`. This is mainly a backup clause. I think accepting < 100% here is okay. If we modify the memory archive to return "None" by default then there's a bad example in code that might be referenced when another plugin is written. Granted having the clause makes it possible in the first place. Not sure what's best here. So I'd leave it as is.